src/poetry/console/commands/run.py

```diff
@@ -44,7 +44,13 @@ def _module(self) -> Module:
 
         return module
 
-    def run_script(self, script: str | dict[str, str], args: str) -> int:
+    def run_script(self, script: str | dict[str, str], args: list[str]) -> int:
+        # Calling `sys.argv` should run the same program as the currently
+        # running program. To make calling Poetry scripts through RunCommand
+        # match this behavior, we must set `sys.argv[0]` to be the full path of
+        # the executable.
+        args = [self.env._bin(args[0]), *args[1:]]
```

Comment:
Not a huge fan of using the `_bin` function here -- could we maybe refactor the `Env` family to have `get_bin` or something else appropriate?

Reply:
Good point. I sticked to `get_bin_path`, it is (almost) self explainable.